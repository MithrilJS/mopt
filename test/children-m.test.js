"use strict";

var code = require("./lib/code");

describe("Children", () => {
    describe("Nested m()", () => {
        it("should support nested m() invocations", () =>
            expect(
                code(`m("0", m("1"))`)
            )
            .toMatchSnapshot()
        );
        
        it("should support an array with an m() invocation", () =>
            expect(
                code(`m("0", [ m("1") ])`)
            )
            .toMatchSnapshot()
        );
        
        it("should support an array with m() invocations", () =>
            expect(
                code(`m("0", [ m("1"), m("2") ])`)
            )
            .toMatchSnapshot()
        );

        it("should support arrays of nested m() invocations & plain nodes", () =>
            expect(
                code(`m("one", [ m("two"), "three" ])`)
            )
            .toMatchSnapshot()
        );
        
        it("should support multiple nested m() invocations", () =>
            expect(
                code(`m("0", m("1"), m("2"), m("3"))`)
            )
            .toMatchSnapshot()
        );
        
        it("should support attrs + nested m() invocations", () =>
            expect(
                code(`m("0", { title : "bar" }, m("1"))`)
            )
            .toMatchSnapshot()
        );
        
        it("should supported attrs + arrays of nested m() invocations", () =>
            expect(
                code(`m("0", { title : "bar" }, [ m("1") ])`)
            )
            .toMatchSnapshot()
        );
        
        it("should support attrs + multiple nested m() invocations", () =>
            expect(
                code(`m("0", { title : "bar" }, m("1"), m("2"), m("3"))`)
            )
            .toMatchSnapshot()
        );
    });
});
