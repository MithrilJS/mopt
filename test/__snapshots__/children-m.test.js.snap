// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children Nested m() should support an array with an m() invocation 1`] = `"m.vnode(\\"0\\",undefined,undefined,[m.vnode(\\"1\\",undefined,undefined,[],undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m() should support an array with m() invocations 1`] = `"m.vnode(\\"0\\",undefined,undefined,[m.vnode(\\"1\\",undefined,undefined,[],undefined,undefined),m.vnode(\\"2\\",undefined,undefined,[],undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m() should support arrays of nested m() invocations & plain nodes 1`] = `"m.vnode(\\"one\\",undefined,undefined,[m.vnode(\\"two\\",undefined,undefined,[],undefined,undefined),m.vnode(\\"#\\",undefined,undefined,\\"three\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m() should support attrs + multiple nested m() invocations 1`] = `"m.vnode(\\"0\\",undefined,{title:\\"bar\\"},[m.vnode(\\"1\\",undefined,undefined,[],undefined,undefined),m.vnode(\\"2\\",undefined,undefined,[],undefined,undefined),m.vnode(\\"3\\",undefined,undefined,[],undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m() should support attrs + nested m() invocations 1`] = `"m.vnode(\\"0\\",undefined,{title:\\"bar\\"},[m.vnode(\\"1\\",undefined,undefined,[],undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m() should support multiple nested m() invocations 1`] = `"m.vnode(\\"0\\",undefined,undefined,[m.vnode(\\"1\\",undefined,undefined,[],undefined,undefined),m.vnode(\\"2\\",undefined,undefined,[],undefined,undefined),m.vnode(\\"3\\",undefined,undefined,[],undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m() should support nested m() invocations 1`] = `"m.vnode(\\"0\\",undefined,undefined,[m.vnode(\\"1\\",undefined,undefined,[],undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m() should supported attrs + arrays of nested m() invocations 1`] = `"m.vnode(\\"0\\",undefined,{title:\\"bar\\"},[m.vnode(\\"1\\",undefined,undefined,[],undefined,undefined)],undefined,undefined);"`;
